dill/_dill.py

```diff
@@ -410,9 +416,69 @@ def save_numpy_array(pickler, obj):
             raise PicklingError(msg)
         logger.trace_setup(self)
         StockPickler.dump(self, obj)
-
     dump.__doc__ = StockPickler.dump.__doc__
 
+    def save(self, obj, save_persistent_id=True, *, name=None):
+        """If self._refonfail is True, try to save object by reference if pickling fails."""
```

Comment:
You shouldn't refer to `self` in the docs -- either use `refonfail is True` or `dill.settings['refonfail'] is True`... basically the doc should refer to how you'd expect a user to set it.  It's better than the `StockPickler.save` docs (i.e. None!), but this is a more complex signature, and I expect it a bit difficult to figure out what all the args do.  For example, I'm going to have to read the code to see what `name` does.  Also, you might want to be more specific when this function is called... is it after any time pickling fails, or just in certain cases (i.e. from certain function calls)?

Reply:
This is "private" API, it's an override of `StockPickler.save()`, which is not documented as you noted. Isn't it better to convert this to a comment? This function is **always** called, once for each object being pickled. And it always calls the superclass method, the only difference being that if `refonfail` is `True` it traps any common pickling exceptions and then it does its real job.

I wrote the last chunks of code in a rush, the documentation is full of gaps. Please keep pointing where the code is in need of minimal explanation. I'll add lines about the parameters.

Reply:
I know it's "private", and that it's always called... my point was that your doc is unclear.  I'd either change it to a comment (to be consistent with `pickle`) and/or improve the docs.

Reply:
I extended/rewrote the documentation for save() and _save_module_dict().